onnxruntime/core/providers/dnnl/graph/primal_nodes/dnnl_primitive_utils.h

```diff
@@ -0,0 +1,293 @@
+        {"Squeeze", "Memory"},
+        // Sum op
+        {"Sum", "Sum"},
+        // Transpose op
+        {"Transpose", "Reorder"},
+        // Unsqueeze op
+        {"Unsqueeze", "Memory"},
+    };
+
+// Maps algorithm name to dnnl::algorithm
+static const OnnxOp2oneDnnAlgorithm onnx_op_to_onednn_algorithm =
+    {
+        // Binary algorithms
+        {"Add",           dnnl::algorithm::binary_add},
+        {"Mul",           dnnl::algorithm::binary_mul},
+        {"Sub",           dnnl::algorithm::binary_sub},
+        {"Div",           dnnl::algorithm::binary_div},
+        // Eltwise algorithms
+        {"Abs",           dnnl::algorithm::eltwise_abs},
+        {"BiasGelu",      dnnl::algorithm::eltwise_gelu_erf},
+        {"Elu",           dnnl::algorithm::eltwise_elu},        // Algorithm requires alpha value
+        {"Equal",         dnnl::algorithm::binary_eq},
+        {"Exp",           dnnl::algorithm::eltwise_exp},
+        {"FastGelu",      dnnl::algorithm::eltwise_gelu_tanh},
+        {"Gelu",          dnnl::algorithm::eltwise_gelu_erf},
+        {"Greater",       dnnl::algorithm::binary_gt},
+        {"GreaterOrEqual",dnnl::algorithm::binary_ge},
+        {"LeakyRelu",     dnnl::algorithm::eltwise_relu},      // Algorithm requires alpha value
+        {"Less",          dnnl::algorithm::binary_lt},
+        {"LessOrEqual",   dnnl::algorithm::binary_le},
+        {"Log",           dnnl::algorithm::eltwise_log},
+        {"Pow",           dnnl::algorithm::eltwise_pow},
+        {"Relu",          dnnl::algorithm::eltwise_relu},
+        {"Round",         dnnl::algorithm::eltwise_round},
+        // OneDNN eltwise_logistic is defined as 1/(1 + exp(-x)) which matches the definition of "Sigmoid" in ONNX
+        {"Sigmoid",       dnnl::algorithm::eltwise_logistic},
+        // OneDNN eltwise_soft_relu is defined as ln(1 + exp(x)) which matches the definition of "Softplus" in ONNX
+        {"Softplus",      dnnl::algorithm::eltwise_soft_relu},
+        {"Sqrt",          dnnl::algorithm::eltwise_sqrt},
+        {"Tanh",          dnnl::algorithm::eltwise_tanh},
+        {"Linear",        dnnl::algorithm::eltwise_linear}
+    };
+
+// Maps dnnl primitive names to the names of the post op compatible primitives
+static const OneDnnRules onednn_postop_rules =
+    {
+        // Add reorder into ops that support implicit casts and format changes
+        {"Convolution",         {"Eltwise", "Sum", "Binary", "Depthwise"}},
+        {"InnerProduct",        {"Eltwise", "Sum", "Binary"}}, 
+        {"MatMul",              {"Eltwise", "Sum", "Binary", "Reorder"}}, 
+        {"BatchNormalization",  {"Eltwise"}},
+        {"Binary",              {"Eltwise", "Sum", "Binary", "Reorder"}},
+        {"Concat",              {}},
+        {"Eltwise",             {"Binary"}},
+        {"LayerNormalization",  {"Eltwise", "Sum", "Binary"}},
+        {"LRN",                 {}},
+        {"Pooling",             {}},
+        {"PReLU",               {}},
```

Comment:
Is this correct: "PReLU" -> "PRelu" ?

Reply:
oneDNN primitive it's called PReLU while ONNX operator is called PRelu, but it's the same, at one point we map the ONNX operators to oneDNN primitives so this should be no issue